src/sentry/integrations/bitbucket/repository.py

```diff
@@ -29,50 +27,16 @@ def get_installation(self, integration_id, organization_id):
 
         return integration_model.get_installation(organization_id)
 
-    def get_config(self, organization):
-        choices = []
-        for i in Integration.objects.filter(organizations=organization, provider='bitbucket'):
-            choices.append((i.id, i.name))
-
-        if not choices:
-            choices = [('', '')]
-        return [
-            {
-                'name': 'integration_id',
-                'label': 'Bitbucket Integration',
-                'type': 'choice',
-                'choices': choices,
-                'initial': choices[0][0],
-                'help': 'Select which Bitbucket integration to authenticate with.',
-                'required': True,
-            },
-            {
-                'name': 'name',
-                'label': 'Repository Name',
-                'type': 'text',
-                'placeholder': 'e.g. getsentry/sentry',
-                'help': 'Enter your repository name, including the owner.',
-                'required': True,
-            },
-        ]
-
     def validate_config(self, organization, config):
-        """
-        ```
-        if config['foo'] and not config['bar']:
-            raise PluginError('You cannot configure foo with bar')
-        return config
-        ```
-        """
-        if config.get('name'):
-            installation = self.get_installation(config['integration_id'], organization.id)
-            client = installation.get_client()
-            try:
-                repo = client.get_repo(config['name'])
-            except Exception as e:
-                installation.raise_error(e)
-            else:
-                config['external_id'] = six.text_type(repo['uuid'])
+        installation = self.get_installation(config['installation'], organization.id)
+        client = installation.get_client()
+        try:
+            repo = client.get_repo(config['identifier'])
+        except Exception as e:
+            installation.raise_error(e)
+        else:
+            config['external_id'] = repo['full_name']
```

Comment:
Should this be the repo id like you had earlier? `config['external_id'] = six.text_type(repo['uuid'])`